### ISSUE TITLE
Incorrect operators used for AND/OR

### ISSUE THREAD

**Issue by miholler** [codecutout/StringToExpression#26]
The ODataFilterLanguage uses bitwise instead of logical AND/OR operators which leads to long running SQL queries, because the conditions are casted as BIT and compared against 0 or 1 instead of checking if the condition is met (True) or not (False).

@alex-davies I would be happy to fix this, but I don't have the permissions to create branches and therefore am not able to create a PR.

**Comment by alex-davies** [comment]
Do you have an example? The dotnet expression generated does use logical and/or, what are you using to convert the expression into SQL?

**Comment by miholler** [comment]
I'm using EF Core to translate an execute the query against the DB but EF Core is working correctly here. 

Imagine having an object with an "Id" and a "Name" property. The following $filter=id eq 1 and name eq 'Foo' would be translated to: x => x.Id == 1 & x.Name == "Foo"
But it should be translated to: x => x.Id == 1 **&&** x.Name == "Foo" (note the double ampersand)

**Comment by alex-davies** [comment]
you are right!!

its weird that this hasn't caused any of the tests to fail and it hasn't been raised earlier

fix is easy, although not quite sure how to catch issue in a unit test. Ill get a fix in tomorrow and hopefully do a new release

**Comment by miholler** [comment]
Thank you very much, that would be great :) For testing, maybe you could write an ExpressionVisitor who checks for the NodeType to be AndAlso/OrElse instead of And/Or?

I would also like to offer you to do a small contribution to your package. I would like to add the StringSyntaxAttribute to the regex patterns of the GrammarDefinitions in order to get syntax highlighting within the IDE for those regex patterns. In my opinion this would be a neat little addition. Furthermore, I could also do a bit of refactoring like removing unused usings, remove empty lines and do a bit of overall code cleanup and formatting if you want me to. Those things could then also go into the next version of the package. Let me know what you think :)

**Comment by alex-davies** [comment]
Should now be resolved in the 2.2.0 release

I took the easier route to testing and checked we got an exception if we tried to do AND/OR operations on integers

`[StringSyntaxAttribute]` is neat, i didn't know that existed. Unfortunately the library at the moment is in netstandard2.0 so cant use the attribute. 

I am part way through some bigger changes to make it easier to turn off and on odata features. That will have to be a major veresion bump so as part of that I was going to update to net8. I can add the attribute in at the same time
